lib/ansible/modules/windows/win_dhcp_lease.ps1

```
@@ -0,0 +1,473 @@
+        if ($type -eq "reservation") {
+            Try {
+                # Update parameters
+                $params = @{ }
+
+                if ($mac) {
+                    $params.ClientId = $mac
+                } else {
+                    $params.ClientId = $current_lease.ClientId
+                }
+
+                if ($description) {
+                    $params.Description = $description
+                } else {
+                    $params.Description = $current_lease.Description
+                }
+
+                if ($reservation_name) {
+                    $params.Name = $reservation_name
+                } else {
+                    $params.Name = "reservation-" + $params.ClientId
+                }
+
+                # Desired type is reservation
+                $current_lease | Add-DhcpServerv4Reservation -WhatIf:$check_mode
+
+                if(-not $check_mode) {
+                    $current_reservation = Get-DhcpServerv4Lease -ClientId $params.ClientId -ScopeId $current_lease.ScopeId
+                }
+
+                # Update the reservation with new values
+                $current_reservation | Set-DhcpServerv4Reservation @params -WhatIf:$check_mode
+
+                if(-not $check_mode) {
+                    $updated_reservation = Get-DhcpServerv4Lease -ClientId $params.ClientId -ScopeId $current_reservation.ScopeId
+                }
+
+                if(-not $check_mode) {
+                    # Compare Values
+                    $module.Result.changed = Compare-DhcpLease -Original $original_lease -Updated $updated_reservation
+                    $module.Result.lease = Convert-ReturnValue -Object $updated_reservation
+                } else {
+                    $module.Result.changed = $true
+                }
+
+                $module.ExitJson()
+            }
+            Catch {
+                $module.Result.changed = $false
+                $module.FailJson("Could not convert lease to a reservation")
```

You should return the error as well

```suggestion
                $module.FailJson("Could not convert lease to a reservation: $($_.Exception.Message)", $_)
```